Tasks/Common/MSBuildHelpers/InvokeFunctions.ps1

```diff
@@ -42,7 +29,16 @@ function Invoke-BuildTools {
                 $splat["LogFileVerbosity"] = $LogFileVerbosity
             }
 
-            Invoke-MSBuild -ProjectFile $file -MSBuildPath $MSBuildLocation -AdditionalArguments $MSBuildArguments -NoTimelineLogger:$NoTimelineLogger @splat
+            if ($Clean) {
```

Comment:
I think a comment on the approach would be helpful:

"If we need to clean, make sure we don't build any additional targets (/t:) twice"

Reply:
With these long command lines, it's hard to see that they're actually the same thing.  I think a DRY approach is clearer:

```pwsh
if ($Clean) {
    Invoke-MSBuild -ProjectFile $file -Targets Clean -MSBuildPath $MSBuildLocation -AdditionalArguments $MSBuildArguments -NoTimelineLogger:$NoTimelineLogger @splat
}

# If we cleaned and passed /t targets, we don't need to run them again
if (!$Clean -or -not $MSBuildArguments.Contains("/t")) {
    Invoke-MSBuild -ProjectFile $file -MSBuildPath $MSBuildLocation -AdditionalArguments $MSBuildArguments -NoTimelineLogger:$NoTimelineLogger @splat
}
```